tests/auth_tests/test_hashers.py

```diff
@@ -457,12 +457,29 @@ def test_argon2(self):
         self.assertTrue(is_password_usable(blank_encoded))
         self.assertTrue(check_password('', blank_encoded))
         self.assertFalse(check_password(' ', blank_encoded))
+        # Old hashes without version attribute
+        encoded = ('argon2$argon2i$m=8,t=1,p=1$c29tZXNhbHQ$gwQOXSNhxiOxPOA0+'
+                   'PY10P9QFO4NAYysnqRt1GSQLE55m+2GYDt9FEjPMHhP2Cuf0nOEXXMoc'
+                   'VrsJAtNSsKyfg')
+        self.assertTrue(check_password('secret', encoded))
+        self.assertFalse(check_password('wrong', encoded))
 
     def test_argon2_upgrade(self):
         self._test_argon2_upgrade('time_cost', 'time cost', 1)
         self._test_argon2_upgrade('memory_cost', 'memory cost', 16)
         self._test_argon2_upgrade('parallelism', 'parallelism', 1)
 
+    def test_argon2_version_upgrade(self):
+        hasher = get_hasher('argon2')
```

Comment:
I would assert for memory/time/parallelism here (and also in the other upgrade test) -- so we know that actually the version caused the upgrade.